### PR TITLE
drop type specs in RecoHI {Configuration, HiTracking}

### PR DESCRIPTION
**Pull request by jeongeun**
#### PR description: 
Update the safer syntax for existing parameter :
- drop type specifications where the original parameter exists.
- move all parameter inside the clone

Instead of modifying parameters with full type specs, which can be interpreted as an insertion of a new parameter, it is a safer way to protect from parameter name mistakes and will also help in possible parameter migrations.
(The previous PR for RecoHI is [PR#32031](https://github.com/cms-sw/cmssw/pull/32031))

In this PR, total 29 files changed. 

- RecoHI/Configuration : 3 files
- RecoHI/HiTracking       : 26 files

#### PR validation:
Event Content comparison check was also done and there is no change with these updates.
Tested in CMSSW_11_2_X, the basic test all passed in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html).

### CODE REVIEW

File: RecoHI/Configuration/python/Reconstruction_hiSignal_cff.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 #Tracks
 hiSignalGlobalPrimTracks = hiGlobalPrimTracks.clone()
 hiSignalSelectedTracks = hiSelectedTracks.clone()
-hiSelectedTracks.src = cms.InputTag("hiSignalGlobalPrimTracks")
+hiSelectedTracks.src = "hiSignalGlobalPrimTracks"
```
Comment (perrotta) [inline RecoHI/Configuration/python/Reconstruction_hiSignal_cff.py:15]: I imagine that the original intention was to modify the cloned `hiSignalSelectedTracks,` and then include in the Task, instead of the `hiSelectedTracks`

File: RecoHI/HiTracking/python/HILowPtConformalPixelTracks_cfi.py [hunk 2]
```diff
@@ -49,78 +49,81 @@
...
-hiConformalPixelTracksPhase1SeedLayers = lowPtQuadStepSeedLayers.clone()
-hiConformalPixelTracksPhase1SeedLayers.BPix = cms.PSet( 
-         HitProducer = cms.string('siPixelRecHits'),
-         TTRHBuilder = cms.string('WithTrackAngle'),
-)
-hiConformalPixelTracksPhase1SeedLayers.FPix = cms.PSet( 
-         HitProducer = cms.string('siPixelRecHits'),
-         TTRHBuilder = cms.string('WithTrackAngle'),
+hiConformalPixelTracksPhase1SeedLayers = lowPtQuadStepSeedLayers.clone(
+    BPix = cms.PSet(
```
Comment (perrotta) [inline RecoHI/HiTracking/python/HILowPtConformalPixelTracks_cfi.py:73]: you can use `dict(...)`

File: RecoHI/HiTracking/python/HILowPtConformalPixelTracks_cfi.py [hunk 3]
```diff
@@ -49,78 +49,81 @@
...
-)
-hiConformalPixelTracksPhase1SeedLayers.FPix = cms.PSet( 
-         HitProducer = cms.string('siPixelRecHits'),
-         TTRHBuilder = cms.string('WithTrackAngle'),
+hiConformalPixelTracksPhase1SeedLayers = lowPtQuadStepSeedLayers.clone(
+    BPix = cms.PSet( 
+	HitProducer = cms.string('siPixelRecHits'),
+        TTRHBuilder = cms.string('WithTrackAngle'),
+    ),
+    FPix = cms.PSet(
```
Comment (perrotta) [inline RecoHI/HiTracking/python/HILowPtConformalPixelTracks_cfi.py:77]: same here

File: RecoHI/HiTracking/python/HILowPtConformalPixelTracks_cfi.py [hunk 4]
```diff
@@ -49,78 +49,81 @@
...
-           enabled = cms.bool(True),
-           pt1 = cms.double(0.7),
-           pt2 = cms.double(2),
-           value1 = cms.double(200),
-           value2 = cms.double(50)
-        )
+    doublets   = "hiConformalPixelTracksPhase1HitDoubletsCA",
+    CAPhiCut   = 0.2,
+    CAThetaCut = 0.0012,
+    SeedComparitorPSet = cms.PSet(
```
Comment (perrotta) [inline RecoHI/HiTracking/python/HILowPtConformalPixelTracks_cfi.py:97]: Perhaps also here (I didn't check...)

File: RecoHI/HiTracking/python/HILowPtConformalPixelTracks_cfi.py [hunk 5]
```diff
@@ -49,78 +49,81 @@
...
-           value2 = cms.double(50)
-        )
+    doublets   = "hiConformalPixelTracksPhase1HitDoubletsCA",
+    CAPhiCut   = 0.2,
+    CAThetaCut = 0.0012,
+    SeedComparitorPSet = cms.PSet( 
+       ComponentName = cms.string('none')
+    ),
+    extraHitRPhitolerance = 0.032,
+    maxChi2 = cms.PSet(
```
Comment (perrotta) [inline RecoHI/HiTracking/python/HILowPtConformalPixelTracks_cfi.py:101]: ...and here

File: RecoHI/HiTracking/python/HIPixelVerticesPreSplitting_cff.py [hunk 6]
```diff
@@ -2,48 +2,62 @@
...
+    medianVertexCollection   = "hiPixelMedianVertexPreSplitting"
 )
 bestHiVertexPreSplittingTask = cms.Task( hiBestAdaptiveVertexPreSplitting , hiSelectedVertexPreSplitting )
 
-PixelLayerTripletsPreSplitting = PixelLayerTriplets.clone()
-PixelLayerTripletsPreSplitting.FPix.HitProducer = 'siPixelRecHitsPreSplitting'
-PixelLayerTripletsPreSplitting.BPix.HitProducer = 'siPixelRecHitsPreSplitting'
-
+PixelLayerTripletsPreSplitting = PixelLayerTriplets.clone(
+    FPix = dict(HitProducer = cms.string('siPixelRecHitsPreSplitting')),
```
Comment (perrotta) [inline RecoHI/HiTracking/python/HIPixelVerticesPreSplitting_cff.py:58]: Why did you need to specify that it is a `cms.string` here?

File: RecoHI/HiTracking/python/HISelectedTracks_cfi.py [hunk 7]
```diff
@@ -5,48 +5,48 @@
...
-    useAnyMVA = cms.bool(True),
-    GBRForestLabel = cms.string('HIMVASelectorIter4'),
-    GBRForestVars = cms.vstring(['chi2perdofperlayer', 'dxyperdxyerror', 'dzperdzerror', 'nhits', 'nlayers', 'eta']),
-    trackSelectors= cms.VPSet(
+    src            = 'hiGlobalPrimTracks',
+    useAnyMVA      = True,
+    GBRForestLabel = 'HIMVASelectorIter4',
+    GBRForestVars  = ['chi2perdofperlayer', 'dxyperdxyerror', 'dzperdzerror', 'nhits', 'nlayers', 'eta'],
+    trackSelectors = cms.VPSet(
     RecoHI.HiTracking.hiMultiTrackSelector_cfi.hiLooseMTS.clone(
```
Comment (perrotta) [inline RecoHI/HiTracking/python/HISelectedTracks_cfi.py:13]: Please indent (here and below)

File: RecoHI/HiTracking/python/hiHighPtTripletStep_cff.py [hunk 8]
```diff
@@ -102,127 +103,127 @@
...
+)
 
 import TrackingTools.KalmanUpdators.Chi2MeasurementEstimator_cfi
 hiHighPtTripletStepChi2Est = TrackingTools.KalmanUpdators.Chi2MeasurementEstimator_cfi.Chi2MeasurementEstimator.clone(
-        ComponentName = cms.string('hiHighPtTripletStepChi2Est'),
-            nSigma = cms.double(3.0),
-            MaxChi2 = cms.double(9.0)# 30 for pp
-        )
+        ComponentName = 'hiHighPtTripletStepChi2Est',
+            nSigma  = 3.0,
```
Comment (perrotta) [inline RecoHI/HiTracking/python/hiHighPtTripletStep_cff.py:120]: indentation

File: RecoHI/HiTracking/python/hiPixelPairStep_cff.py [hunk 9]
```diff
@@ -26,52 +26,58 @@
...
-            layerList = cms.vstring('BPix1+BPix2', 'BPix1+BPix3', 'BPix2+BPix3',
-                                    'BPix1+FPix1_pos', 'BPix1+FPix1_neg',
-                                    'BPix2+FPix1_pos', 'BPix2+FPix1_neg',
-                                    'FPix1_pos+FPix2_pos', 'FPix1_neg+FPix2_neg'),
-            BPix = cms.PSet(
+    layerList = ['BPix1+BPix2', 'BPix1+BPix3', 'BPix2+BPix3',
+                 'BPix1+FPix1_pos', 'BPix1+FPix1_neg',
+                 'BPix2+FPix1_pos', 'BPix2+FPix1_neg',
+                 'FPix1_pos+FPix2_pos', 'FPix1_neg+FPix2_neg'],
+    BPix = cms.PSet(
```
Comment (perrotta) [inline RecoHI/HiTracking/python/hiPixelPairStep_cff.py:33]: `dict()` ?

File: RecoHI/HiTracking/python/hiPixelPairStep_cff.py [hunk 10]
```diff
@@ -26,52 +26,58 @@
...
         TTRHBuilder = cms.string('TTRHBuilderWithoutAngle4PixelPairs'),
         HitProducer = cms.string('siPixelRecHits'),
         skipClusters = cms.InputTag('hiPixelPairClusters')
-        ),
-            FPix = cms.PSet(
+    ),
+    FPix = cms.PSet(
```
Comment (perrotta) [inline RecoHI/HiTracking/python/hiPixelPairStep_cff.py:38]: `dict()` ?

File: RecoHI/HiTracking/python/hiRegitInitialStep_cff.py [hunk 11]
```diff
@@ -23,22 +23,24 @@
...
-hiRegitInitialStepSeeds.RegionFactoryPSet                                           = HiTrackingRegionFactoryFromJetsBlock.clone()
-hiRegitInitialStepSeeds.ClusterCheckPSet.doClusterCheck                             = False # do not check for max number of clusters pixel or strips
-hiRegitInitialStepSeeds.skipClusters = cms.InputTag('hiRegitInitialStepClusters')
-hiRegitInitialStepSeeds.RegionFactoryPSet.RegionPSet.ptMin = 1.2
-
+hiRegitInitialStepSeeds     = RecoTracker.IterativeTracking.InitialStep_cff.initialStepSeeds.clone(
+    RegionFactoryPSet = HiTrackingRegionFactoryFromJetsBlock.clone(),
+    ClusterCheckPSet = dict(doClusterCheck = False), # do not check for max number of clusters pixel or strips
+    skipClusters = cms.InputTag('hiRegitInitialStepClusters')
+    RegionFactoryPSet = dict(
```
Comment (perrotta) [inline RecoHI/HiTracking/python/hiRegitInitialStep_cff.py:40]: This can be moved inside the `clone` above

File: RecoHI/HiTracking/python/hiRegitLowPtTripletStep_cff.py [hunk 12]
```diff
@@ -19,31 +19,38 @@
...
+hiRegitLowPtTripletStepSeeds     = RecoTracker.IterativeTracking.LowPtTripletStep_cff.lowPtTripletStepSeeds.clone(
+    RegionFactoryPSet = HiTrackingRegionFactoryFromJetsBlock.clone(),
+    ClusterCheckPSet = dict(doClusterCheck = False), # do not check for max number of clusters pixel or strips
+    OrderedHitsFactoryPSet = dict(
+	SeedingLayers = 'hiRegitLowPtTripletStepSeedLayers',
+	GeneratorPSet = dict (
+	    SeedComparitorPSet = RecoPixelVertexing.PixelLowPtUtilities.LowPtClusterShapeSeedComparitor_cfi.LowPtClusterShapeSeedComparitor.clone()
+	),
+    ),
+    RegionFactoryPSet = dict(
```
Comment (perrotta) [inline RecoHI/HiTracking/python/hiRegitLowPtTripletStep_cff.py:50]: This can be moved inside the `clone()` above

File: RecoHI/HiTracking/python/hiRegitMixedTripletStep_cff.py [hunk 13]
```diff
@@ -19,54 +19,57 @@
...
-hiRegitMixedTripletStepSeedsA.RegionFactoryPSet                                           = HiTrackingRegionFactoryFromJetsBlock.clone()
-hiRegitMixedTripletStepSeedsA.ClusterCheckPSet.doClusterCheck                             = False # do not check for max number of clusters pixel or strips
-hiRegitMixedTripletStepSeedsA.OrderedHitsFactoryPSet.SeedingLayers = 'hiRegitMixedTripletStepSeedLayersA'
-hiRegitMixedTripletStepSeedsA.RegionFactoryPSet.RegionPSet.ptMin = 1.0
-
+hiRegitMixedTripletStepSeedsA = RecoTracker.IterativeTracking.MixedTripletStep_cff.mixedTripletStepSeedsA.clone(
+    RegionFactoryPSet = HiTrackingRegionFactoryFromJetsBlock.clone(),
+    ClusterCheckPSet = dict(doClusterCheck = False), # do not check for max number of clusters pixel or strips
+    OrderedHitsFactoryPSet = dict(SeedingLayers = 'hiRegitMixedTripletStepSeedLayersA'),
+    RegionFactoryPSet = dict(
```
Comment (perrotta) [inline RecoHI/HiTracking/python/hiRegitMixedTripletStep_cff.py:49]: This can be moved inside the `clone()` above

File: RecoHI/HiTracking/python/hiRegitMixedTripletStep_cff.py [hunk 14]
```diff
@@ -19,54 +19,57 @@
...
+hiRegitMixedTripletStepSeedLayersB =  RecoTracker.IterativeTracking.MixedTripletStep_cff.mixedTripletStepSeedLayersB.clone(
+    BPix = dict(skipClusters = 'hiRegitMixedTripletStepClusters'),
+    TIB  = dict(skipClusters = 'hiRegitMixedTripletStepClusters'),
+    layerList = ['BPix2+BPix3+TIB1','BPix2+BPix3+TIB2']
+)
+hiRegitMixedTripletStepSeedsB = RecoTracker.IterativeTracking.MixedTripletStep_cff.mixedTripletStepSeedsB.clone(
+    RegionFactoryPSet = HiTrackingRegionFactoryFromJetsBlock.clone(),
+    ClusterCheckPSet = dict(doClusterCheck = False), # do not check for max number of clusters pixel or strips
+    OrderedHitsFactoryPSet = dict(SeedingLayers = 'hiRegitMixedTripletStepSeedLayersB'),
+    RegionFactoryPSet = dict(
```
Comment (perrotta) [inline RecoHI/HiTracking/python/hiRegitMixedTripletStep_cff.py:63]: This can be moved inside the `clone()` above

File: RecoHI/HiTracking/python/hiRegitPixelPairStep_cff.py [hunk 15]
```diff
@@ -19,30 +19,30 @@
...
-hiRegitPixelPairStepSeeds.RegionFactoryPSet                                           = HiTrackingRegionFactoryFromJetsBlock.clone()
-hiRegitPixelPairStepSeeds.ClusterCheckPSet.doClusterCheck                             = False # do not check for max number of clusters pixel or strips
-hiRegitPixelPairStepSeeds.OrderedHitsFactoryPSet.SeedingLayers = 'hiRegitPixelPairStepSeedLayers'
-hiRegitPixelPairStepSeeds.RegionFactoryPSet.RegionPSet.ptMin = 1.2
-
+hiRegitPixelPairStepSeeds     = RecoTracker.IterativeTracking.PixelPairStep_cff.pixelPairStepSeeds.clone(
+    RegionFactoryPSet = HiTrackingRegionFactoryFromJetsBlock.clone(),
+    ClusterCheckPSet = dict(doClusterCheck = False), # do not check for max number of clusters pixel or strips
+    OrderedHitsFactoryPSet = dict(SeedingLayers = 'hiRegitPixelPairStepSeedLayers'),
+    RegionFactoryPSet = dict(RegionPSet = dict(ptMin = 1.2)),
```
Comment (perrotta) [inline RecoHI/HiTracking/python/hiRegitPixelPairStep_cff.py:44]: This can be moved inside the `clone()` above

**Comment by perrotta** [comment]
please test

**Comment by perrotta** [comment]
> Comparison job queued.

@smuzaffar , I have the impression that in the last few days the message "Comparison job queued." even in cases, as this one, when the testing procedure has ended (in this case because of build errors). Is it normal?

**Comment by smuzaffar** [comment]
@perrotta , thanks for reporting, This should be fixed now ( https://github.com/cms-sw/cms-bot/commit/2732c257d425a1a1150ab31964edc3ebbb0b6347 )

**Comment by perrotta** [comment]
> @perrotta , thanks for reporting, This should be fixed now ( [cms-sw/cms-bot@69a8d33](https://github.com/cms-sw/cms-bot/commit/69a8d333a7aad80eac6451299146071628109f9a) )

Great! Thank you @smuzaffar

**Comment by jeongeun** [comment]
Close this PR and this update will be on new PR.
